trin-beacon/src/storage.rs

```diff
@@ -0,0 +1,532 @@
+                }
+            }
+        }
+    }
+
+    /// The "radius: concept is not applicable for Beacon network, this is why we always return the
+    /// max radius.
+    fn radius(&self) -> Distance {
+        Distance::MAX
+    }
+}
+
+impl BeaconStorage {
+    pub fn new(config: PortalStorageConfig) -> Result<Self, ContentStoreError> {
+        let metrics = StorageMetricsReporter {
+            storage_metrics: PORTALNET_METRICS.storage(),
+            protocol: ProtocolId::Beacon.to_string(),
+        };
+        let storage = Self {
+            node_data_dir: config.node_data_dir,
+            sql_connection_pool: config.sql_connection_pool,
+            storage_capacity_in_bytes: config.storage_capacity_mb * BYTES_IN_MB_U64,
+            metrics,
+            network: ProtocolId::Beacon,
+            cache: BeaconStorageCache::new(),
+        };
+
+        // Report current storage capacity.
+        storage
+            .metrics
+            .report_storage_capacity_bytes(storage.storage_capacity_in_bytes as f64);
+
+        // Report current total storage usage.
+        let total_storage_usage = storage.get_total_storage_usage_in_bytes_on_disk()?;
+        storage
+            .metrics
+            .report_total_storage_usage_bytes(total_storage_usage as f64);
+
+        // Report total storage used by network content.
+        let network_content_storage_usage =
+            storage.get_total_storage_usage_in_bytes_from_network()?;
+        storage
+            .metrics
+            .report_content_data_storage_bytes(network_content_storage_usage as f64);
+
+        Ok(storage)
+    }
+
+    fn db_insert(
+        &self,
+        content_id: &[u8; 32],
+        content_key: &String,
+        value: &Vec<u8>,
+    ) -> Result<(), ContentStoreError> {
+        let conn = self.sql_connection_pool.get()?;
+        insert_value(conn, content_id, content_key, value, u8::from(self.network))
+    }
+
+    fn db_insert_lc_update(&self, period: &u64, value: &Vec<u8>) -> Result<(), ContentStoreError> {
+        let conn = self.sql_connection_pool.get()?;
+        let value_size = value.len();
+
+        match conn.execute(
+            INSERT_LC_UPDATE_QUERY,
+            params![period, value, 0, value_size],
+        ) {
+            Ok(_) => Ok(()),
+            Err(err) => Err(err.into()),
+        }
+    }
+
+    pub fn store(
+        &mut self,
+        key: &impl OverlayContentKey,
+        value: &Vec<u8>,
+    ) -> Result<(), ContentStoreError> {
+        let content_id = key.content_id();
+        let content_key: Vec<u8> = key.clone().into();
+
+        match content_key.first() {
+            Some(&LIGHT_CLIENT_BOOTSTRAP_KEY_PREFIX) => {
+                // store content key w/o the 0x prefix
+                let content_key = hex_encode(content_key).trim_start_matches("0x").to_string();
+                if let Err(err) = self.db_insert(&content_id, &content_key, value) {
```

Comment:
No, because we don't have a "capacity" for the beacon network. Every node should store all of the beacon data (it is not too much) and we don't use radius and we don't leave an option for the user to decide how much mbs she wants to allocate to this network.

Reply:
Well... isn't it possible that the pruning inside the `HistoryStorage` will prune beacon data? Which is not something we want to happen, if I'm understanding the situation correctly

Reply:
`HistoryStorage` prunes only history data because `find_farthest_content_id` uses `XOR_FIND_FARTHEST_QUERY` which requires a `network` param to look into the main db.